src/components/administerOrder.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import {connect} from 'react-redux';
+
+import {getOrderForAdministration, foodOrdered} from '../store/ordersActions';
+import OrderDetails from './orderDetails';
+import Chat from './chat';
+import OrderState from '../enums/orderState';
+import messageBuilderFactory from './messageBuilderFactory';
+
+const AdministerOrder = React.createClass({
+    propTypes: {
+        dispatch: React.PropTypes.func.isRequired,
+        order: React.PropTypes.object.isRequired,
+        params: React.PropTypes.object.isRequired
+    },
+
+    componentDidMount () {
+        this.props.dispatch(getOrderForAdministration(this.props.params.adminId));
+    },
+
+    renderActions () {
+        const {state: orderState} = this.props.order;
+
+        switch (orderState) {
+        case (OrderState.Open):
+            return (
+                <button
+                    className="btn btn-primary"
+                    onClick={() => this.props.dispatch(foodOrdered(this.props.params.adminId))}
+                >Food ordered</button>
+            );
+        default:
+            return null;
+        }
+    },
+
+    render () {
+        const {order} = this.props;
+
+        const orderReady = () => order.id;
```

Comment:
I suppose when not loaded order is '{}' - an empty object? I was wondering - is it a correct way to do it with redux? shouldn't it be null?